File: app/src/main/java/com/github/fredrik9000/firmadetaljer_android/repository/room/CompanyDao.java
package com.github.fredrik9000.firmadetaljer_android.repository.room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface CompanyDao {

    @Insert
    void insert(Company company);

    @Delete
    void delete(Company company);

    @Query("SELECT * FROM company_table ORDER BY id DESC")
    LiveData<List<Company>> getAllCompaniesOrderedByLastInserted();
}

File: app/src/main/java/com/github/fredrik9000/firmadetaljer_android/repository/room/CompanyDao.kt
package com.github.fredrik9000.firmadetaljer_android.repository.room

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface CompanyDao {

    @get:Query("SELECT * FROM company_table ORDER BY id DESC")
    val allCompaniesOrderedByLastInserted: LiveData<List<Company>>

    @Insert
    suspend fun insert(company: Company)

    @Query("DELETE FROM company_table WHERE organisasjonsnummer = :organisasjonsnummer")
    suspend fun deleteByOrganizationNumber(organisasjonsnummer: Int)

    @Query("DELETE FROM company_table")
    suspend fun deleteAll()
}
